class.jetpack-jitm.php

```diff
@@ -135,9 +150,40 @@ function manage_pi_msg() {
 		if ( current_user_can( 'jetpack_manage_modules' ) ) {
 			$normalized_site_url = Jetpack::build_raw_urls( get_home_url() );
 			$manage_active = Jetpack::is_module_active( 'manage' );
-			$manage_pi_dismissed = isset( self::$jetpack_hide_jitm['manage-pi'] );
+			// If it's not an array, it means no JITM was dismissed
+			$manage_pi_dismissed = isset( self::$jetpack_hide_jitm['manage-pi'] ) || is_array( self::$jetpack_hide_jitm );
+			// Check if plugin has auto update already enabled in WordPress.com and don't show JITM in such case.
```

Comment:
I think it would be useful to break this logic off into it's own boolean method so it could be used elsewhere.